spec/javascript/testUtils/wrapperUtil.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+import Context from '~/components/supplemental_application/context'
+import { Form } from 'react-final-form'
+
+const formNode = (application, formToChildrenFunc) => (
+  <Form
+    onSubmit={() => null}
+    initialValues={application}
+    render={({ form }) => (
+      <form noValidate>
+        {formToChildrenFunc(form)}
+      </form>
+    )}
+  />
+)
+
+const diveThroughFormWrappers = (wrapper) =>
+  wrapper
+    .find('form')
+    .children()
+    .dive()
+
+const shallowWithForm = (application, formToChildrenFunc) =>
+  diveThroughFormWrappers(
+    shallow(
+      formNode(application, formToChildrenFunc)
+    )
+  )
+
+const mountWithForm = (application, formToChildrenFunc) =>
+  mount(formNode(application, formToChildrenFunc))
+
+/**
+ * Return a component wrapper for any component that uses react-final-form
+ *
+ * @param application the application object to seed the form with
+ * @param formToChildrenFunc a function that takes a form object and returns a node.
+ * @param shouldMount true if component should be rendered with enzyme mount.
+ *  Only use shouldMount=true if you know you absolutely need mount functionality.
```

Comment:
could be helpful to clarify here or in the readme docs what we expect could be a good reason for using mount. I'm still not clear on it myself.

Reply:
I added a short explanation here and a long explanation in the readme, want to take a look and see if it makes sense to you?

I'm heading offline for the night btw

Reply:
yes I think it's good!